docs/core/resilience/http-resilience.md

```diff
@@ -0,0 +1,321 @@
+            {
+                // Imagine a/b testing, where 3% of the requests are 
+                // sent to the experimental endpoint.
+                new() { Uri = new("https://example.com/experimental"), Weight = 3 },
+                new() { Uri = new("https://example.com/stable"), Weight = 97 }
+            }
+        });
+    });
+
+    builder.ConfigureWeightedGroups(static weightedGroup =>
+    {
+        // Alternatively, you can use initial attempt as the selection mode.
+        weightedGroup.SelectionMode =
+            WeightedGroupSelectionMode.EveryAttempt;
+
+        weightedGroup.Groups.Add(new WeightedEndpointGroup()
+        {
+            Endpoints =
+            {
+                // Apply 33% of the requests to each endpoint
+                new() { Uri = new("https://example.com/a"), Weight = 33 },
+                new() { Uri = new("https://example.com/b"), Weight = 33 },
+                new() { Uri = new("https://example.com/c"), Weight = 33 }
+            }
+        });
+    });
+});
+```
+
+The preceding code:
+
+- Adds the hedging handler to the <xref:Microsoft.Extensions.DependencyInjection.IHttpClientBuilder>.
+- Configures the `IRoutingStrategyBuilder` to use the `ConfigureOrderedGroups` method to configure the ordered groups.
+- Adds an `EndpointGroup` to the `orderedGroup` that routes 3% of the requests to the `https://example.com/experimental` endpoint and 97% of the requests to the `https://example.com/stable` endpoint.
+- Configures the `IRoutingStrategyBuilder` to use the `ConfigureWeightedGroups` method to configure the weighted groups.
+- Sets the `SelectionMode` to `WeightedGroupSelectionMode.EveryAttempt`.
+- Adds a `WeightedEndpointGroup` to the `weightedGroup` that routes 33% of the requests to the `https://example.com/a` endpoint, 33% of the requests to the `https://example.com/b` endpoint, and 33% of the requests to the `https://example.com/c` endpoint.
+
+For more information, see [Polly docs: Hedging resilience strategy](https://www.pollydocs.org/strategies/hedging.html).
+
+## Add custom resilience handlers
+
+To have more control, you can customize the resilience handlers by calling the `AddResilienceHandler` extension method. This method takes a delegate that configures the `ResiliencePipelineBuilder<HttpResponseMessage>` instance that is used to create the resilience strategies.
+
+To configure a named resilience handler, call the `AddResilienceHandler` extension method with the name of the handler. The following example configures a named resilience handler called `"CustomPipeline"`.
+
+```csharp
+// The builder is of type, IHttpClientBuilder.
+builder.AddResilienceHandler(
+    "CustomPipeline",
+    static (ResiliencePipelineBuilder<HttpResponseMessage> builder,
+            ResilienceHandlerContext context) =>
+{
+    // See: https://www.pollydocs.org/strategies/retry.html
+    builder.AddRetry(new HttpRetryStrategyOptions
+    {
+        BackoffType = DelayBackoffType.Exponential,
```

Comment:
Maybe here we can also include a sample to configure what responses and exceptions are retried by setting-up the `ShouldHandle` predicate?

You can use either switch expresions or `PredicateBuilder<HttpResponseMessage>.`

Reply:
I was showing an example of that below.